### DISCUSSION TITLE
Extracting individual models from stacking regressor

### DISCUSSION THREAD

**Discussion by Sukantabasu** [microsoft/FLAML#488] [Q&A]
I am using "ensemble" = True and "model_history" = True. I would like to extract individual models from the following stacking regressor. What is the procedure? 
 
StackingRegressor(estimators=[('lgbm',
                               <flaml.model.LGBMEstimator object at 0x153e400a41c0>),
                              ('xgboost',
                               <flaml.model.XGBoostSklearnEstimator object at 0x153e400a4220>),
                              ('xgb_limitdepth',
                               <flaml.model.XGBoostLimitDepthEstimator object at 0x153e400a4250>),
                              ('catboost',
                               <flaml.model.CatBoostEstimator object at 0x153e400a4280>),
                              ('rf',
                               <flaml.model.RandomForestEstimator object at 0x153e400a42b0>),
                              ('extra_tree',
                               <flaml.model.ExtraTreesEstimator object at 0x153e400a42e0>)],
                  n_jobs=-1, passthrough=True)

**Comment by sonichi** [comment]
StackingRegressor object has an attribute "estimators_", which a list of estimators. Is that what you are looking for?
`automl.model.estimators_`

**Comment by Sukantabasu** [comment]
No, I would like to load individually trained sub-models (best lgb, best rf, etc.). I would like to make predictions from each one of them separately for comparison.

**Reply by sonichi** [reply]
These two attributes can be used:
https://microsoft.github.io/FLAML/docs/reference/automl#best_model_for_estimator
https://microsoft.github.io/FLAML/docs/reference/automl#best_loss_per_estimator

**Reply by Sukantabasu** [reply]
Thanks a lot for the information!

**Reply by sonichi** [reply]
You're welcome. I'd appreciate your feedback in a survey: https://docs.google.com/forms/d/e/1FAIpQLSd0u4BbSzr-m9hgM8rmPlYB_JJoFXeHhuoXI9YFbHDRWrDlOQ/viewform?usp=sf_link.

**Reply by Sukantabasu** [reply]
Done...

**Reply by sonichi** [reply]
@Sukantabasu Thanks for your feedback. I have some followup questions for you. Do you mind joining on [gitter](https://gitter.im/FLAMLer/community)? Or if there is an alternative way to contact you, could you please let me know? Thank you very much.

**Comment by Sukantabasu** [comment]
Hi Chi, 

You can contact me at ***@***.*** 

Best regards,
Sukanta

-- 
Sukanta Basu
Associate Professor
Faculty of Civil Engineering and Geosciences
Delft University of Technology

T: +31-15-27 89358 (office)
T: +31-6-15557518 (cell)

https://sites.google.com/view/sukantabasu/
https://www.researchgate.net/profile/Sukanta_Basu
https://scholar.google.com/citations?user=08bv9p8AAAAJ&hl=en
----------------------------------------------------------
